File: python/helpers/mqtt_edge_auth.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for
# license information.
import datetime
import ssl
from typing import Any


class MqttEdgeAuth(object):
    def __init__(self) -> None:
        self.token_expiry: datetime.datetime = None
        self.client_id: str = None
        self.username: str = None
        self.password: str = None
        self.hostname: str = None
        self.server_verification_cert: str = None
        self.device_id: str = None
        self.module_id: str = None
        self.port: int = 8883

    @classmethod
    def create_from_environment(
        cls, token_renewal_interval: int = 3600, token_renewal_margin: int = 300
    ) -> Any:
        """
        create a new Edge auth object from the environment.

        :param int token_renewal_interval: Number of seconds that SAS tokens created by
            this obhject will be valid.
        :param int token_renewal_margen: Number of seconds to subtract from
            `token_renewal_inteavral` when calculating the `token_expiry` time.

        :returns: MqttEdgeAuth object created by this function.
        """
        pass

    def renew_token(self) -> None:
        """
        Renew authorization. This casuses a new password string to be generated.
        """
        pass

    def create_tls_context(self) -> ssl.SSLContext:
        """
        Create an SSLContext object based on this object.

        :returns: SSLContext object
        """
        pass

File: node/helpers/mqtt_edge_auth.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

const PORT = 8888;
const HOST = 'host';
const KEY = 'path_to_key';
const CERT = 'path_to_cert';
const TRUSTED_CA_LIST = 'list';

interface MqttEdgeAuthParams {
  port : number;
  host : string;
  key : string;
  cert : string;
  rejectUnauthorized : boolean;
  ca : string;
  deviceId : string;
  moduleId : string;
}

export class MqttEdgeAuth {
  port : number;
  host : string;
  key : string;
  cert : string;
  rejectUnauthorized : boolean;
  ca : string;
  deviceId : string;
  moduleId : string;

  constructor(input: MqttEdgeAuthParams) {
    this.port = input.port;
    this.host = input.host;
    this.key = input.key;
    this.cert = input.cert;
    this.rejectUnauthorized = true;
    this.ca = input.ca;
    this.deviceId = input.deviceId;
    this.moduleId = input.moduleId;
  }

  static createFromEnvironment() {
    return new MqttEdgeAuth({
      port: PORT,
      host: HOST,
      key: KEY,
      cert: CERT,
      rejectUnauthorized: true,
      ca: TRUSTED_CA_LIST,
      moduleId: 'moduleId',
      deviceId: 'deviceId'
    }) ;
  }
}